src/utils.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Hasher;
+
+pub struct Utils {}
+
+impl Utils {
+  pub fn digest_b64_url_only_ascii(hasher: &dyn Hasher, input: &str) -> String {
+    // "The digest MUST be taken over the US-ASCII bytes of the base64url-encoded Disclosure".
+    let ascii_bytes: Vec<u8> = input.as_bytes().iter().cloned().filter(|&byte| byte <= 127).collect();
+    let hash = hasher.digest(&ascii_bytes);
+    // "The bytes of the digest MUST then be base64url-encoded".
+    multibase::Base::Base64Url.encode(hash)
```

Comment:
> The digest MUST be taken over the US-ASCII bytes of the base64url-encoded Disclosure 

in the spec sounded to me like non-ASCII should just be filtered out. But you're right. I'll fix it. https://github.com/iotaledger/sd-jwt/issues/3

Reply:
What they mean is what encoding (i.e. US-ASCII) of the string should be used when base64-encoding it. It could be any number of other weird encodings that encode the base64 string (representing the disclosure) represented with different bytes than ASCII does.